docs/01-app/01-getting-started/03-layouts-and-pages.mdx

```diff
@@ -247,6 +247,139 @@ export default async function BlogPostPage({ params }) {
+  const fromHook = useParams()
+
+  return (
+    <div>
+      <p>{fromProps.slug}</p>
+      <p>{fromHook.slug}</p>
+    </div>
+  )
+}
+```
+
+```jsx filename="app/blog/[slug]/page.js" switcher
+'use client'
+import { use } from 'react'
+import { useParams } from 'next/navigation'
+
+export default function BlogPostPage({ params }) {
+  const fromProps = use(params)
+  const fromHook = useParams()
+
+  return (
+    <div>
+      <p>{fromProps.slug}</p>
+      <p>{fromHook.slug}</p>
+    </div>
+  )
+}
+```
+
+## Rendering with search params
+
+### `searchParams`
+
+In a Server Component **page**, you can access search parameters using the `searchParams` prop:
+
+```tsx filename="app/page.tsx" switcher
+export default async function Page({
+  searchParams,
+}: {
+  searchParams: Promise<{ [key: string]: string | string[] | undefined }>
+}) {
+  const { from } = await searchParams
+
+  // validate that `from` can be parsed into a valid date
+
+  // query your database for records `from`
+
+  return <div>{/* render records */}</div>
+}
+```
+
+```jsx filename="app/page.jsx" switcher
+export default async function Page({ searchParams }) {
+  const { from } = await searchParams
+
+  // validate that `from` can be parsed into a valid date
+
+  // query your database for records `from`
+
+  return <div>{/* render records */}</div>
+}
+```
+
+Using `searchParams` opts your page into [**dynamic rendering**](/docs/app/getting-started/partial-prerendering#dynamic-rendering) because it requires a incoming request to read the search parameters from.
+
+Client Component **pages** can also access `searchParams` using React’s `use` hook:
+
+```tsx filename="app/page.tsx" switcher
+'use client'
+import { use } from 'react'
+
+export default function Page({
+  searchParams,
+}: {
+  searchParams: Promise<{ [key: string]: string | string[] | undefined }>
+}) {
+  const { from } = use(searchParams)
+
+  // validate that `from` can be parsed into a valid date
+
+  // query your database for records `from`
+
+  return <div>{/* render records */}</div>
+}
+```
+
+```jsx filename="app/page.jsx" switcher
+'use client'
+import { use } from 'react'
+
+export default function Page({ searchParams }) {
+  const { from } = use(searchParams)
+
+  // validate that `from` can be parsed into a valid date
+
+  // query your database for records `from`
+
+  return <div>{/* render records */}</div>
+}
+```
+
+Verify that the incoming request has [access rights](/docs/app/guides/authentication) to these records.
+
+### `useSearchParams`
+
+In a [**dynamically rendered**](/docs/app/getting-started/partial-prerendering#dynamic-rendering) page, Client Components can use the [`useSearchParams`](/docs/app/api-reference/functions/use-search-params) hook to read search parameters, without being opted-out from being server-side rendered. Learn more about [Client Components server side rendering](/docs/app/getting-started/server-and-client-components#on-the-server).
```

Comment:
I think a guide showing patterns of `searchParams` being used for serve side "state", vs `useSearchParams` used to drive client side operations, among other things is needed.

This is a getting started page, to get familiar with what APIs are available, and their basic principles.